roles/openshift_health_checker/openshift_checks/existing_ovs_version.py

```diff
@@ -0,0 +1,68 @@
+# pylint: disable=missing-docstring
+"""
+Ansible module for determining if an installed version of Open vSwitch is incompatible with the
+currently installed version of OpenShift.
+"""
+
+from openshift_checks import OpenShiftCheck, OpenShiftCheckException, get_var
+
+
+class ExistingOvsVersion(OpenShiftCheck):
+
+    name = "existing_ovs_version"
+    tags = ["health"]
+
+    openshift_to_ovs_version = {
+        "3.6": "2.6",
+        "3.5": "2.6",
+        "3.4": "2.4",
+    }
+
+    @classmethod
+    def is_active(cls, task_vars):
+        """Skip hosts that do not have package requirements."""
+        group_names = get_var(task_vars, "group_names", default=[])
+        is_master_or_node = 'masters' in group_names or 'nodes' in group_names
+        return super(ExistingOvsVersion, cls).is_active(task_vars) and is_master_or_node
+
+    def run(self, tmp, task_vars):
+        args = {
+            "package_list": [
+                (
+                    "openvswitch",
+                    self.get_required_ovs_version(task_vars),
+                ),
+            ],
+        }
+        return self.execute_module("rpm_version", args, task_vars)
+
+    def get_required_ovs_version(self, task_vars):
+        """Return the correct Open vSwitch version for the current OpenShift version.
+        If the current OpenShift version is >= 3.5, ensure Open vSwitch version 2.6,
+        Else ensure Open vSwitch version 2.4"""
```

Comment:
We don't need to describe the implementation details in a docstring. As soon as the implementation changes, the documentation would easily become wrong.